lib/features/support/servlet.rb

```diff
@@ -67,6 +68,35 @@ def log_request(request)
       $logger.debug "BODY: #{JSON.pretty_generate(JSON.parse(request.body))}"
     end
   end
+
+  def check_digest(request)
+    header = request['Bugsnag-Integrity']
+    return if header.nil?
```

Comment:
I wonder if we should have a flag that can be set in an after configuration hook, which would tell Maze Runner that a missing integrity header is an error (or even whether it should check the header at all)

The implicit checking is nice, but if it silently does nothing when the header isn't present then it feels a bit dangerous

e.g.

```ruby
AfterConfiguration do
  MazeRunner.config.check_integrity_header = true
end
```

```diff
+    return unless MazeRunner.config.check_integrity_header
+
     header = request['Bugsnag-Integrity']
-    return if header.nil?
+    raise 'No Bugsnag-Integrity header found' if header.nil?
```

Reply:
I'm not against that, but if we did that then I'd prefer it got passed in as a command line option.  The `config` construction is more a by-product of that and more for internal MR use that for clients to interact with.

Also, in terms of checking that a header is there once implemented by the notifier, I saw that as where the notifier should have its own scenarios to explicitly check for it (as Android has).

Reply:
...that is, the intention of the implicit checking was an additional safeguard on top of what notifier should be checking for themselves, as opposed to it being our only safety net.

Reply:
Yeah I get it, but it's a bit of a shame to have to add assertions to the notifier to know that it's being checked. This definitely doesn't block merging, but I think it'd be nice to opt in to a stricter mode eventually

About config vs command line options; I'd prefer leaning towards configuring MR in ruby if it's something that is universal to all MR tests, e.g. if we added a command line option to the Ruby pipeline we'd need to edit 71 different steps — not hard to do, but it's easy to miss some or forget to add it to new steps in the future

Reply:
OK, I've raised a fresh internal ticket (PLAT-5419) to look at that.  Good point about accessing config directly at the language level if it works for a given notifier, I just wouldn't want to force other platforms down that route.  So we can add an option and the entry in `MazeRunner.config` will happen naturally.